apps/server-asset-sg/src/features/assets/search/asset-search.service.ts

```diff
@@ -161,17 +162,19 @@ export class AssetSearchService {
    * Searches for assets using a {@link AssetSearchQuery}.
    *
    * @param query The query to match with.
+   * @param user The user whose assets are to be searched.
    * @param limit The maximum amount of assets to load. Defaults to `100`.
    * @param offset The amount of assets being skipped before loading the assets.
    * @param decode Whether to decode the assets. If this is set to `false`, the assets should be decoded via `AssetEditDetail` before accessing them.
    *               This option primarily exists so that the assets are not decoded and directly re-encoded when returning them via API.
    */
   async search(
     query: AssetSearchQuery,
+    user: User,
```

Comment:
I would leave it like this for now and go over it again in #163. It might even be possible to fully move the user/its workgroups into the query, so that we don't need the extra parameter.